File: src/Domestica.Budget.Application/Transactions/AddIncomeTransaction/AddIncomeTransactionCommand.cs
using CommonAbstractions.DB.Messaging;
using Domestica.Budget.Domain.Transactions;

namespace Domestica.Budget.Application.Transactions.AddIncomeTransaction
{
    public sealed record AddIncomeTransactionCommand(
        string DestinationAccountId,
        string SenderName,
        decimal TransactionAmount,
        string CategoryValue,
        string SubcategoryValue,
        DateTime TransactionDateTime) : ICommand<Transaction>
    {
    }
}

File: src/client-app/src/models/requests/addIncomeTransactionCommand.ts
export interface AddIncomeTransactionCommand {
    destinationAccountId: string
    senderName: string
    transactionAmount: number
    categoryValue: string
    transactionDateTime: Date
}

export class AddIncomeTransactionCommand implements AddIncomeTransactionCommand {
    constructor(values?: AddIncomeTransactionCommand) {
        this.destinationAccountId = values?.destinationAccountId || ''
        this.senderName = values?.senderName || ''
        this.transactionAmount = values?.transactionAmount || 0
        this.categoryValue = values?.categoryValue || ''
        this.transactionDateTime = values?.transactionDateTime || new Date()
    }
}